src/cargo/core/source/source_id.rs

```diff
@@ -44,6 +44,8 @@ enum Kind {
     Path,
     /// represents a remote registry
     Registry,
+    /// represents a remote alternative registry
+    AltRegistry,
```

Comment:
I'd ideally like to avoid having a separate source for alternative registries here, would it be possible to just use `Registry` above and recognize crates.io through an identifier like "crates-io"? Either that or perhaps a method like `is_default_registry(&self) -> bool` or something like that.

Reply:
The problem is that in order to distinguish if it is the default registry in unit tests it becomes more of a pain as we can't purely check for if the index matches crates.io index. One change that I made as part of #4697 was to store the name in the source_id, that would then allow just checking if that was present, rather than there being an additional type, would you prefer that approach?

Reply:
I updated this to use the name as that also fixed the bug that you highlighted.